crates/router/src/core/payments/helpers.rs

```diff
@@ -1885,29 +1887,84 @@ pub async fn retrieve_card_with_permanent_token(
         .change_context(errors::ApiErrorResponse::UnprocessableEntity {
             message: "no customer id provided for the payment".to_string(),
         })?;
+
+    let db = state.store.as_ref();
+    let key_manager_state = &state.into();
+    let merchant_account = db
+        .find_merchant_account_by_merchant_id(
+            key_manager_state,
+            &payment_intent.merchant_id,
+            merchant_key_store,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
+    if merchant_account.is_network_tokenization_enabled {
+        let pm_data = db
+            .find_payment_method(payment_method_id, merchant_account.storage_scheme)
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)?;
+        if let Some(mandate_ids) = mandate_id {
+            if let Some(api_models::payments::MandateReferenceId::NetworkTokenWithNTI(nt_data)) =
+                mandate_ids.mandate_reference_id
+            {
+                if let Some(network_token_locker_id) = pm_data.network_token_locker_id {
+                    let mut token_data = cards::get_card_from_locker(
+                        state,
+                        customer_id,
+                        &payment_intent.merchant_id,
+                        network_token_locker_id.as_ref(),
+                    )
+                    .await
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable(
+                        "failed to fetch card information from the permanent locker",
```

Comment:
if this fails, we should try fetching the card details right? we should log the error and then move forward